### ISSUE TITLE
New players not stored in cmi_users mysql table

### ISSUE THREAD

**Issue by digitalcrowd** [Zrips/CMI#7014]
Background: All was running good until May 1, when I updated numerous plugins including CMI to latest versions at that time, latest Paper version, etc. Didn't notice any problems, until on June 1, I went to update all my plugins to their latest versions and restarted the server, **one player reported they lost all their set homes**. No one else reported any issues and I dismissed it as a fluke.

A few days ago, I upgraded my server to 1.19 and updated all my plugins and restarted the server. The same player reported they lost all their set homes again. This was followed by another player having the same issue. I also notice their balance was reset to the default balance level.

Being a database guy and programmer (not java) I did some digging around and noticed that no new players since May 1 had been entered into the cmi_users table. Players who were in the table already, were not impacted and had successful updates to the table for their set homes changes, balance, etc.

Using CoreProtect plugin I was able to track down 93 players which didn't have any data in cmi_users since May 1. While I realize that potentially, another plugin may be interfering with, I think it would be good to have some checks and balances, that if your plugin goes to update the user table for a player, if there is an error doing an UPDATE to the table, that, it should INSERT an entry for the player to correct the issue.

As a test, the one player in question (mentioned above), I sent ahead and inserted some simple data for him into the cmi_users table with his UUID and playername. I had him connect to the server and he was on for a little bit, but no changes to the table, then I had him disconnect. Within a minute or less of him disconnecting from the server, the cmi_users table updated his information, storing all his set homes (that apparently where being stored in memory) properly, etc.

So I went int and inserted the other 93 players into the cmi_users table, so that when they connect and disconnect, it will save properly for them.

Of course, I am wondering, and this has nothing to do with this issue, but is your plugin storing player data in memory the entire time until restart, regardless if the player is connected or not, or did it just store the data in memory because of the error upon mysql update. This seems like big overhead of memory that would be unnecessary. I may got 30 days without restarting my server (or longer), if there have been no recent big updates to anything.

It's hard to troubleshoot this issue, since I would need a handful of new player accounts to do troubleshooting with that have not connected to my server, since I have no clue at what point CMI inserts new player data or insert new player data for players whom are not new, but would be "new" to CMI. If I knew that, then I could just delete a test player account data I have from the database, and then reconnect to see if indeed it is another plugin causing the issue or not.

---

#### DELETE THIS SECTION IF YOU ARE SUBMITTING A FEATURE REQUEST ####

**ERROR (DELETE IF YOU HAVE NO ERROR):**

No errors in the log.

####################
####################
```

**CONFIG SECTION (DELETE IF NOT RELEVANT):**

Not relevant because it works fine to connect to the DB if the player is already in the DB.

#####################################
## PASTE RELEVANT CONFIG SECTION HERE ##
#####################################

Not needed. No configuration changes happened between when I updated the plugins on May 1 and when I started having issues.

---

**Cmi Version (using`/cmi version`):** 

[14:26:01 INFO]: CMI plugin version: 9.2.0.3 MySQL-> 9.2.0.4
[14:26:01 INFO]: CMILib: 1.2.0.2 -> 1.2.0.3
[14:26:01 INFO]: Server: Paper(36) 1.19-R0.1-SNAPSHOT
[14:26:01 INFO]: CMI economy: Enabled Vault: 1.7.3-b CMI Chat: Enabled
[14:26:01 INFO]: Modules -> 53 enabled 3 disabled: flightCharge, disabledEnchants, namePlates


**Server Type (Spigot/Paperspigot/etc):** 

**Server Version (using `/ver`):** 

[14:26:27 INFO]: This server is running Paper version git-Paper-36 (MC: 1.19) (Implementing API version 1.19-R0.1-SNAPSHOT) (Git: 5593f4e)
You are 3 version(s) behind
Download the new version at: https://papermc.io/downloads
Previous version: git-Paper-33 (MC: 1.19)

**Comment by andrew121410** [comment]
Yes, all users are stored in memory constantly even if they're offline.

**Comment by digitalcrowd** [comment]
Okay - Good to know on the memory thing.

What about the data not being stored in the cmi_users for new players?

**Comment by Zrips** [comment]
@digitalcrowd 
Without any errors its hard to say why it would not save players data as its not widespread issue, so some specific setup causes this issue. Do you have any option to test this without any other plugin? Try running with sqlite database type? You could try changing database name, in case you have multiple servers running on same tables and overriding its information. Or even make a copy and try running it on your local PC. Additionally you could try running default config files with only MySQL data base being setup.
Can't really say much here from my end why it would fail saving
